src/java.base/share/classes/sun/security/pkcs12/PKCS12KeyStore.java

```diff
@@ -1958,10 +1958,23 @@ public synchronized void engineLoad(InputStream stream, char[] password)
         certPbeIterationCount = -1;
         macAlgorithm = null;
         macIterationCount = -1;
+        String keystorePath = null;
 
         if (stream == null)
            return;
 
+        if (debug != null) {
+            if (stream instanceof FileInputStream) {
+                keystorePath = SharedSecrets
+                                .getJavaIOFileInputStreamAccess()
+                                .getPath((FileInputStream) stream);
+                if (keystorePath != null) {
+                    debug.println("PKCS12KeyStore: Loading \"" + keystorePath.substring(
```

Comment:
I ran a straightforward test to connect to a TLS website and expected to see the cacerts file details being printed. 

`BufferedInputStream` is used for `InputStream` instance

sun.security.util.KeyStoreDelegator#engineLoad

```
            InputStream bufferedStream = new BufferedInputStream(stream);
            bufferedStream.mark(Integer.MAX_VALUE);

            try {
                @SuppressWarnings("deprecation")
                KeyStoreSpi tmp = primaryKeyStore.newInstance();
                tmp.engineLoad(bufferedStream, password);
```

You'll need to get a handle to the BufferedInputStream -> FilterInputStream -> `in` stream and check if it's an instance of a `FileInputStream`. You can then query and print the path value if present.

On a side note, I was surprised to see no debug output (of interest) initially when I was using the `-Djava.security.debug=keystore` value. Turns out I have to use the `pkcs12` value for debug option. Given that PKCS12KeyStore.java is the only file to use this debug value and that pkcs12 is now the default keystore type, there might be merit in folding this value into the more common `keystore` option